### PR TITLE
python-websockets: new package

### PR DESCRIPTION
**Pull request by paper42**
Maintainer: me
Compile tested: armv7, Turris Omnia, OpenWrt 21.02
Run tested: armv7, Turris Omnia, OpenWrt 21.02

Description: A python library for websocket servers and clients.

### CODE REVIEW

File: lang/python/python-websockets/Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=python-websockets
+PKG_VERSION:=10.0
+PKG_RELEASE:=1
```
Comment (commodo) [inline lang/python/python-websockets/Makefile:12]: any objections against $(AUTORELEASE) ?
also fine to leave this as-is

**Review by commodo — APPROVED** [review]
looks good :)

File: lang/python/python-websockets/Makefile (hunk 1 above)
Comment (paper42) [reply lang/python/python-websockets/Makefile]: No, I just forgot about it. Thank you for noticing this :+1:

**Comment by jefferyto** [comment]
Please run:

```sh
make package/python-websockets/{clean,configure} PY3=stdlib V=s
```

and take note of the output.

**Comment by paper42** [comment]
> Please run:
> 
> ```shell
> make package/python-websockets/{clean,configure} PY3=stdlib V=s
> ```
> 
> and take note of the output.

If I run this command, I get an error:
```
/usr/bin/env bash ../python3-find-stdlib-depends.sh -n "python-websockets" "/home/michal/nic/build-omnia-hbd/build/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/pypi/websockets-10.0"  if grep -q "websockets==" ../host-pip-requirements/*.txt ; then if ! grep -q "websockets==10.0" ../host-pip-requirements/*.txt ; then printf "\nPlease update version of websockets to 10.0 in 'host-pip-requirements'/\n\n" ; exit 1 ; fi fi
bash: -c: line 1: syntax error near unexpected token `then'
bash: -c: line 1: `/usr/bin/env bash ../python3-find-stdlib-depends.sh -n "python-websockets" "/home/michal/nic/build-omnia-hbd/build/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/pypi/websockets-10.0"        if grep -q "websockets==" ../host-pip-requirements/*.txt ; then if ! grep -q "websockets==10.0" ../host-pip-requirements/*.txt ; then printf "\nPlease update version of websockets to 10.0 in 'host-pip-requirements'/\n\n" ; exit 1 ; fi fi'
```

relevant section of python3-package.mk:

```make
define Py3Build/FindStdlibDepends
	$(SHELL) $(python3_mk_path)python3-find-stdlib-depends.sh -n "$(PKG_NAME)" "$(PKG_BUILD_DIR)"
endef

ifneq ($(strip $(PYPI_NAME)),)
define Py3Build/CheckHostPipVersionMatch
	if grep -q "$(PYPI_NAME)==" $(python3_mk_path)host-pip-requirements/*.txt ; then \
		if ! grep -q "$(PYPI_NAME)==$(PKG_VERSION)" $(python3_mk_path)host-pip-requirements/*.txt ; then \
			printf "\nPlease update version of $(PYPI_NAME) to $(PKG_VERSION) in 'host-pip-requirements'/\n\n" ; \
			exit 1 ; \
		fi \
	fi
endef
endif
```

If I add a `;` to the end of the body of the first define, it starts working. This sounds like a bug, but I am not sure why only I noticed it.

I added the parts of the python package it gave me to the Makefile, thank you.

**Comment by commodo** [comment]
> If I add a `;` to the end of the body of the first define, it starts working. This sounds like a bug, but I am not sure why only I noticed it.

my bad for this;
i am not sure why this worked for me;
maybe i did not run this command?

can you add a fix to this PR?

**Comment by commodo** [comment]
> > If I add a `;` to the end of the body of the first define, it starts working. This sounds like a bug, but I am not sure why only I noticed it.
> 
> my bad for this; i am not sure why this worked for me; maybe i did not run this command?
> 
> can you add a fix to this PR?

or if @jefferyto or someone else prefers another PR, that also works;

**Comment by jefferyto** [comment]
I don't mind if it is a separate commit in the same PR :smile:

**Comment by paper42** [comment]
> can you add a fix to this PR?

> I don't mind if it is a separate commit in the same PR

commit added

File: lang/python/python3-package.mk [hunk 2]
```diff
@@ -201,7 +201,7 @@ PYTHON3_PKG_HOST_PIP_INSTALL_ARGS = \
 	)
 
 define Py3Build/FindStdlibDepends
-	$(SHELL) $(python3_mk_path)python3-find-stdlib-depends.sh -n "$(PKG_NAME)" "$(PKG_BUILD_DIR)"
+	$(SHELL) $(python3_mk_path)python3-find-stdlib-depends.sh -n "$(PKG_NAME)" "$(PKG_BUILD_DIR)";
```
Comment (commodo) [inline lang/python/python3-package.mk:204]: hmm; are you sure that `CheckHostPipVersionMatch` isn't the one thatt needs to be fixed?

Comment (commodo) [reply lang/python/python3-package.mk:204]: actually, never mind;
this is good as-is here;

it seems that when i added `Py3Build/CheckHostPipVersionMatch` the concatenation of  shell commands requires this semi-colon here;

but it only occurs when `PY3=stdlib` is set

**Comment by commodo** [comment]
`cryptodev-linux` still complains....
oh well

**Pull request merged**